### PR TITLE
registration: Split test into many small ones

### PR DESCRIPTION
**Pull request by JohanLorenzo**
Not a lot of changes done, except the tests now use `stainless` and are broken down into 4. r? @hfiguiere

### CODE REVIEW

**Comment by JohanLorenzo** [comment]
Rebased on top of #89. IPv4 and IPv6 tests are not under 2 distinct submodules. I also used regex to validate `registrar.get_ip_addr()`. There, I didn't handle the case where we only have ipv6 addresses. I'm not sure how urgent it is to implement this particular check

r? @hfiguiere

File: src/registration.rs [hunk 1]
```diff
@@ -138,115 +138,130 @@ impl Registrar {
...
+                .unwrap();
+            assert_eq!(ip_addr, "::ffff:192.168.0.4");
+        }
+
+        it "should return IPv4 if both are specified" {
+            let ip_addr = registrar.get_ip_addr_from_ifaces(&interfaces, &Some("eth0".to_owned()))
+                .unwrap();
+            assert_eq!(ip_addr, "192.168.0.4");
+        }
+    }
```
Comment (hfiguiere) [inline src/registration.rs:265]: what about testing `registrar.get_ip_addr_from_ifaces(&interfaces, &None)` ? This was tested before.

Comment (JohanLorenzo) [inline src/registration.rs:265]: I'm not too sure. Isn't that already covered by the combination of these 2 cases:
- ["should default to eth0"](https://github.com/fxbox/foxbox/pull/103/files#diff-81c55a5b1db198d473ae06791f9953ffR198)
- and ["should return IPv4 if both are specified"](https://github.com/fxbox/foxbox/pull/103/files#diff-81c55a5b1db198d473ae06791f9953ffR261)

Comment (hfiguiere) [inline src/registration.rs:265]: in the first case there is no IPv6 listed in the interfaces...

Comment (hfiguiere) [inline src/registration.rs:265]: since you removed it from the sample data.

**Comment by hfiguiere** [comment]
Feel free to land this if you want. I'll address the test coverage issues afterwards.

File: src/registration.rs (hunk 1 above)
Comment (JohanLorenzo) [inline src/registration.rs:265]: Done.

**Comment by JohanLorenzo** [comment]
> Feel free to land this if you want. I'll address the test coverage issues afterwards.

[Test added](https://github.com/fxbox/foxbox/pull/103/files#diff-81c55a5b1db198d473ae06791f9953ffR261).

**Pull request merged**